### PR TITLE
build.js: support flags

### PR DESCRIPTION
**Pull request by tomasmatus**
Adds flags to build.js to use rsync, disable linting and use watch mode.

### CODE REVIEW

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Can you please update README accordingly?

**Comment by tomasmatus** [comment]
> Thanks! Can you please update README accordingly?

ah yes, will do!

File: build.js [hunk 1]
```diff
@@ -14,13 +14,20 @@ import { esbuildStylesPlugins } from './pkg/lib/esbuild-common.js';
...
 import { stylelintPlugin } from './pkg/lib/esbuild-stylelint-plugin.js';
 
+const production = process.env.NODE_ENV === 'production';
 const useWasm = os.arch() !== 'x64';
 const esbuild = (await import(useWasm ? 'esbuild-wasm' : 'esbuild')).default;
 
-const production = process.env.NODE_ENV === 'production';
-const watchMode = process.env.ESBUILD_WATCH === "true";
-// linters dominate the build time, so disable them for production builds by default, but enable in watch mode
-const lint = process.env.LINT ? (process.env.LINT !== '0') : (watchMode || !production);
```
Comment (jelly) [inline build.js:23]: I thought we supported this still in `cockpit` but just checked and it seems not!.

**Comment by martinpitt** [comment]
I'm not sure how many people actually use the env vars. `LINT=0 make watch` would have been a sensible invocation, but I'm okay with letting this go.

File: build.js (hunk 1 above)
Comment (jelly) [reply build.js:23]: But shouldn't we keep `LINT` to be flexible in Makefile's @martinpitt what do you think?

**Comment by tomasmatus** [comment]
Maybe then I can keep both options possible? Env variables and build.js flags

**Comment by martinpitt** [comment]
Yes, but maybe let's see if anyone actually complains :grin:

**Review by jelly — CHANGES_REQUESTED** [review]
This line still uses `ESLINT` needs updating https://github.com/cockpit-project/starter-kit/blob/3d75eb66d7559c5cf2f761f061147382af2f72d0/packaging/cockpit-starter-kit.spec.in#L35

File: README.md [hunk 2]
```diff
@@ -110,7 +110,11 @@ During fast iterative development, you can also choose to not run eslint/styleli
 This speeds up the build and avoids build failures due to e. g. ill-formatted
 css or other issues:
 
-    $ make LINT=0
+    $ ./build.js -e -s
```
Comment (martinpitt) [inline README.md:113]: Let's do this as `-es` for short, as humans will actually type this.

File: README.md [hunk 3]
```diff
@@ -110,7 +110,11 @@ During fast iterative development, you can also choose to not run eslint/styleli
...
+
+or
+
+    $ LINT=0 make
```
Comment (martinpitt) [inline README.md:117]: Not a strong opinion, but let's drop this. Cockpit doesn't support it either, and it's nicer to have *one* way of doing things.

File: build.js [hunk 4]
```diff
@@ -14,13 +14,20 @@ import { esbuildStylesPlugins } from './pkg/lib/esbuild-common.js';
...
+const parser = (await import('argparse')).default.ArgumentParser();
+parser.add_argument('-r', '--rsync', { help: "rsync bundles to ssh target after build", metavar: "HOST" });
+parser.add_argument('-w', '--watch', { action: 'store_true', help: "Enable watch mode", default: process.env.ESBUILD_WATCH === "true" });
+parser.add_argument('-e', '--no-eslint', { action: 'store_true', help: "Disable eslint linting", default: process.env.LINT ? process.env.LINT === '0' : production });
+parser.add_argument('-s', '--no-stylelint', { action: 'store_true', help: "Disable stylelint linting", default: process.env.LINT ? process.env.LINT === '0' : production });
```
Comment (martinpitt) [inline build.js:25]: Very long lines, and DRY. Please keep it outside as a const (like in old line 23) and rename it to `lint_default`.

File: packaging/cockpit-starter-kit.spec.in [hunk 5]
```diff
@@ -32,7 +32,7 @@ rm -rf dist
 %endif
 
 %build
-ESLINT=0 NODE_ENV=production make
+LINT=0 NODE_ENV=production make
```
Comment (martinpitt) [inline packaging/cockpit-starter-kit.spec.in:35]: Ah, this is a good justification for keeping `$LINT`.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

**Review by martinpitt — APPROVED** [review]
Very nice, thank you! Can you please also apply this to all our other projects, so that we don't keep getting confused?

**Comment by martinpitt** [comment]
@jelly you still have a -1 finger on this, can you PTAL?

**Comment by tomasmatus** [comment]
> Very nice, thank you! Can you please also apply this to all our other projects, so that we don't keep getting confused?

Yes, I will do that once this is merged

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
Nice, now podman/machines can have it as well \o/

**Pull request merged**